prow/plugins/trigger/generic-comment.go

```diff
@@ -121,6 +147,12 @@ func handleGenericComment(c Client, trigger plugins.Trigger, gc github.GenericCo
 	if err != nil {
 		return err
 	}
+
+	addedComment, err := addHelpComment(c.GitHubClient, len(toTest), len(toSkip), gc.Body, org, repo, branch, pr.Number, presubmits, gc.HTMLURL, commentAuthor, c.Logger)
```

Comment:
We should not call a method that decides if it should or should do something -- decide here and conditionally add the help comment.

Reply:
Thank you for the review. Fixed.